src/gps/RTC.cpp

```diff
@@ -113,6 +113,12 @@ bool perhapsSetRTC(RTCQuality q, const struct timeval *tv, bool forceUpdate)
         // Every 12 hrs we will slam in a new GPS or Phone GPS / NTP time, to correct for local RTC clock drift
         shouldSet = true;
         LOG_DEBUG("Reapplying external time to correct clock drift %ld secs\n", tv->tv_sec);
+    } else if (q >= RTCQualityFromNet && tv->tv_sec > (zeroOffsetSecs + 3 * 24 * 60 * 60)) {
```

Comment:
Maybe this would make the upgrade safer? so it only tries to fix if we have a clock of low quality

```suggestion
    } else if (currentQuality == RTCQualityFromNet && tv->tv_sec > (zeroOffsetSecs + 3 * 24 * 60 * 60)) {
```